internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -324,6 +324,7 @@ func (r *Reconciler) computeDesiredMachineSet(ctx context.Context, deployment *c
 	} else {
 		desiredMS.Spec.DeletePolicy = ""
 	}
+	desiredMS.Spec.Template.Spec.ReadinessGates = deployment.Spec.Template.Spec.ReadinessGates
```

Comment:
Might be worth to do one quick test with TIlt to ensure in-place propagation of this field works and we're not missing anything

Eventually we can add test coverage by adding something that doesn't lead to behavioral changes to readinessGates in clusterclass_rollout.go ~ l.226. But for now we can't because we are not adding this field to Cluster.spec.topology and ClusterClass. Can we add a point about Cluster.spec.topology, CC & corresponding e2e test coverage to the umbrella issue please?

Reply:
yep, I'm testing testing it
The other part is already in the tracking issue